### PR TITLE
41 userdata object

### LINKED ISSUE #41: userData object

### PR DESCRIPTION
**Pull request by bendsp**
The goal is to create a singe context hook in which the most essential and most widely used information can be stored and updated dynamically

### CODE REVIEW

**Comment by bendsp** [comment]
the updating works now but I'm not super happy with the design - will need to review.

File: types/userData.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    provider?: string;
+}
+
+type Privy = {
+    user?: _privy_io_public_api.PrivyUser;
+    wallet?: EmbeddedWalletState;
+    provider?: PrivyEmbeddedWalletProvider;
+}
+
+export type { User, Wallet, Privy };
```
Comment (PaulThblt) [inline types/userData.ts]:
```suggestion
export type { User, Wallet, Privy };

```

Comment (PaulThblt) [reply types/userData.ts]: T'as pu régler ton linter?

File: components/TestModule.tsx [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
 import FText from './Text/FText';
+import alchemy from '../services/alchemyService';
+
+import { useConfig } from '~/components/Wrappers/UserContext';
+import signMessage from '~/services/privyService';
+import viem from '~/services/viemService';
 
 const TestModule = () => {
-  const user = usePrivy();
+  const Puser = usePrivy();
```
Comment (PaulThblt) [inline components/TestModule.tsx]: So do we still need to use this?
Shouldn't this be inside the userContext?

File: components/Wrappers/UserContext.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
+// UserContext.tsx
+import * as _privy_io_public_api from '@privy-io/public-api';
```
Comment (PaulThblt) [inline components/Wrappers/UserContext.tsx]: Is this used somewhere in the file?

**Comment by bendsp** [comment]
UserData has become AppData and is finally ready to merge to dev

**Comment by PaulThblt** [comment]
Love it, however, `CreateWalletButton` is outdated but I guess it will be removed either way with our new login

**Review by PaulThblt — APPROVED** [review]

**Pull request merged**